src/sources/util/tcp.rs

```diff
@@ -60,7 +55,8 @@ fn make_listener(
     }
 }
 
-pub trait TcpSource: Clone + Send + 'static {
+// TODO: add async_trait for making `run` async.
+pub trait TcpSource: Clone + Send + Sync + 'static {
```

Comment:
Should we make `run` method in this trait async? cc @lukesteensen

Reply:
It will need to be made async eventually, though that would probably be worth a separate PR unless it makes this one significantly easier.

Reply:
Sure, separate PR. Just wanted know the opinion in general.